internal/checker/checker.go

```diff
@@ -64,44 +68,88 @@ func (c *checker) Start() {
 		if err != nil {
 			log.Printf("Error fetching message: %v\n", err)
 		}
-		for _, url := range message.Value.URLs {
-			err := c.ErrorBasedCheck(url)
-			if err != nil {
-				log.Printf("Error-based check error:%v\n", err)
-			}
+		result := c.processMessage(message)
+		result.Result = deleteEmpty(result.Result)
+		if len(result.Result) == 0 {
+			result.Result = append(result.Result, "SQLi-check: No vulnerabilities found.")
+		}
+		err = c.sendResult(result)
+		if err != nil {
+			log.Println(err)
+		} else {
+			log.Printf("Results have been successfully sent to Result Collector.")
+		}
+
+	}
+}
+
+// processMessage starts checking given list of url, appending results received.
+func (c *checker) processMessage(m broker.Message) *result.Result {
+	results := make([]string, 0)
+	totalResult := result.Result{TaskID: m.Value.ID, Result: make([]string, 0)}
+
+	for _, url := range m.Value.URLs {
+		res, err := c.ErrorBasedCheck(url)
+		if err != nil {
+			log.Printf("Error-based check:%v\n", err)
 		}
-		// set it to nil for stop processing previous result while no messages from Kafka.
+		results = append(results, res...)
+	}
+	totalResult.Result = append(totalResult.Result, results...)
+	return &totalResult
+}
+
+// sendResult sends given results via gRPC.
+func (c *checker) sendResult(r *result.Result) error {
+	conn, err := grpc.Dial(c.config.GRPCConfig.ResultCollectorAddress, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
+	if err != nil {
+		return fmt.Errorf("Cannot connect to gRPC server: %w\n", err)
 	}
+	defer conn.Close()
+	client := result.NewCheckerClient(conn)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	_, err = client.SendResult(ctx, &result.ResultRequest{ResultData: r})
+	if err != nil {
+		return fmt.Errorf("Error sending result %v\n", err)
+	}
+
+	return nil
 }
 
 // ErrorBasedCheck firstly reads payloads from the file, then creates regexp for errors, after gets site body
 // checking it on possible inclusions of error keywords, finally posts form on site with injectable parameters.
 // Having this done matches response body with possible error keywords.
-func (c *checker) ErrorBasedCheck(link string) error {
+func (c *checker) ErrorBasedCheck(link string) ([]string, error) {
+	results := make([]string, 0)
 	// reading file with payloads with path from config.
 	f, err := os.Open(c.config.Checker.ErrBasedPayload)
```

Comment:
I think reading the file on each run is not a really good idea. You can move it to main.

Reply:
There are a lot of payloads in the file. I'm not sure about hold then in memory all the time. However it's a really nice question which approach is more suitable here.